src/sentry/api/endpoints/sentry_app_details.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import absolute_import
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.base import Endpoint, SessionAuthentication
+from sentry.api.serializers import serialize
+from sentry.constants import SentryAppStatus
+from sentry.models import SentryApp
+
+
+class SentryAppDetailsEndpoint(Endpoint):
+    authentication_classes = (SessionAuthentication, )
+    permission_classes = (IsAuthenticated, )
+
+    def get(self, request, sentry_app_id):
+        if not any(features.has('organizations:internal-catchall', org)
+                   for org in request.user.get_orgs()):
+            return Response(status=404)
+
+        try:
+            sentry_app = SentryApp.objects.get(id=sentry_app_id)
+        except SentryApp.DoesNotExist:
+            return Response(status=404)
+
+        # Superusers have access to the app, published or unpublished. Other
+        # users only have access to a published app (for now)
+        if request.user.is_superuser or sentry_app.status == SentryAppStatus.PUBLISHED:
```

Comment:
Should this logic live in a permission class? The `has_object_permissions` and `has_permission` hooks would map well to the permissions logic you have for list and details end points so far.

Reply:
Hmm, yea, probably. Some of this is going to come down to the role you have within an Org too, which also probably makes sense to be in the permissions class. 👍